cairo/ethereum/cancun/vm/instructions/environment.cairo

```diff
@@ -439,3 +460,272 @@ func codecopy{range_check_ptr: felt, evm: Evm}() -> ExceptionalHalt* {
     let ok = cast(0, ExceptionalHalt*);
     return ok;
 }
+
+// @notice Get the code size of an external contract
+func extcodesize{
+    range_check_ptr, poseidon_ptr: PoseidonBuiltin*, bitwise_ptr: BitwiseBuiltin*, evm: Evm
+}() -> ExceptionalHalt* {
+    alloc_locals;
+
+    // STACK
+    let stack = evm.value.stack;
+    with stack {
+        let (address_u256, err) = pop();
+        if (cast(err, felt) != 0) {
+            return err;
+        }
+    }
+
+    // GAS
+    let accessed_addresses = evm.value.accessed_addresses;
+    let accessed_addresses_ptr = cast(accessed_addresses.value.dict_ptr, DictAccess*);
+    tempvar address_u256_ = UnionUintU256(new UnionUintU256Enum(cast(0, Uint*), address_u256));
+    let address_ = to_address(address_u256_);
+    tempvar address = new Address(address_.value);
+    let (is_present) = hashdict_read{dict_ptr=accessed_addresses_ptr}(1, &address.value);
+    if (is_present == 0) {
+        // If the entry is not in the accessed storage keys, add it
+        hashdict_write{dict_ptr=accessed_addresses_ptr}(1, &address.value, 1);
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar accessed_addresses_ptr = accessed_addresses_ptr;
```

Comment:
worth checking whether this explicit rebind is required here because they should already be at ap-2 ap-1

Reply:
unforntunately it is revoked if not rebinded